model/src/main/kotlin/Package.kt

```diff
@@ -58,6 +58,12 @@ data class Package(
      */
     val declaredLicenses: SortedSet<String>,
 
+    /**
+     * The list of authors declared for this package.
+     */
+    @JsonInclude(JsonInclude.Include.NON_DEFAULT)
+    val declaredAuthors: SortedSet<String> = sortedSetOf(),
```

Comment:
I believe what @mnonnenmacher meant when he said "Instead of adding a default value here, you could already pass an empty set in all package managers" was that you should not only not assign `null` here, but no default value at all. That way you're forced to initialize the field in all package managers, which IMO was the intent. And that also implies that you should squash the two commits.

Also, as for licenses, we IMO should always serialize the field, even if it's empty.

Finally, I'm a bit unsure about the field ordering. My hunch is that we should keep all license-related fields grouped together, and add the new `declaredAuthors` between `concludedLicense` and `description`. What do you think, @mnonnenmacher?

Reply:
> 
> 
> I believe what @mnonnenmacher meant when he said "Instead of adding a default value here, you could already pass an empty set in all package managers" was that you should not only not assign `null` here, but no default value at all. That way you're forced to initialize the field in all package managers, which IMO was the intent. 

Introducing this new field without a default value is a rather breaking change because then existing ORT results can no longer be deserialized. This would also impact the data stored in scan result storages. This may be acceptable, but is something we should be aware of.

Reply:
My bad, I didn't think of deserializing old results. @mnonnenmacher I guess we want / need to maintain backward-compatibility here? But we should still drop `@JsonInclude(JsonInclude.Include.NON_DEFAULT)`, right?

Reply:
The annotation is not necessary. We put it here for practical reasons because otherwise all tests for analyzers would fail that compare current results against stored yml files. Maybe we could keep the annotation until we have reworked the analyzer implementations to populate the authors field? In the course of this change the test files would have to be adapted anyway.

Reply:
> Finally, I'm a bit unsure about the field ordering.

I grouped `declaredLicenses` and `declaredAuthors` in order to have to the "declared" things together. 

> But we should still drop @JsonInclude(JsonInclude.Include.NON_DEFAULT), right?

As @oheger-bosch already mentioned, if we drop the annotation here then we have to adapted all analyzer tests in this PR and again after we have reworked the analyzers. Can we keep the annotation here with a `TODO` comment e.g. "Can be removed after all package manager implementations have filled the field declaredAuthors accordingly."?

Reply:
I agree with that suggestion.

Reply:
> I grouped `declaredLicenses` and `declaredAuthors` in order to have to the "declared" things together.

But why is `declaredAuthors` then between `declaredLicenses` and `declaredLicensesProcessed` instead of after `declaredLicensesProcessed`?

Reply:
Here's one more thought: Does it even make sense to call this `declaredAuthors` instead of just `authors` after all? While our original idea of naming the field `declaredCopyrights` made sense to distinguish from *detected* copyrights, we already figured that this field does not contain full copyright statements in the "detected copyright" sense as it lacks the actual "Copyright (C)" prefix and the year.

So why not treat this as just another metadata field, like e.g. `homepageUrl`, name it just `authors`, put it before `description`, and let later logic consider "authors" as "declared copyright holders" (similar to like we also consider the "homepageUrl" later as a fallback for a VCS URL)?

Reply:
The original plan was to treat author information analogously to declared licenses and allow an analogous curation mechanism. In this context, the name `declaredAuthors` would make sense.

If we treat authors as regular metadata, would we then still want to curate them? Or would the curation rather apply to the logic that transforms authors to copyright holders? If so, would the curations still be defined in `curations.yml` or would this be a different mechanism?

Reply:
Please have a look at the [PackageCurationData](https://github.com/oss-review-toolkit/ort/blob/master/model/src/main/kotlin/PackageCurationData.kt) class: We can curate any metadata, no matter whether it start with a "declared" prefix or not. So my point is purely about how to name the field, the original plan and all processing of the field's data stays the same.

My (somewhat rhetorical) question boils down to: If we had a `declaredAuthors` field, what does "declared" mean here, and how does it semantically differ from just `authors`, as *any* metadata in the `Package` / `Project` classes is just "declared" (and not detected) metadata. Phrased differently, why is `homepageUrl` *not* called `declaredHomepageUrl`. Well, because "declared" is redundant in *this* (the analyzer's) context.

Going further off-topic: Consequently, I actually believe we should have named the `declaredLicenses` fields also just `licenses` here, and only further tooling that also has to deal with other kinds of licenses should refer to it as `declaredLicenses`.

Reply:
> But why is declaredAuthors then between declaredLicenses and declaredLicensesProcessed instead of after declaredLicensesProcessed?

Because for me `declaredLicensesProcessed` is not a field for gathering metadata directly from the package manager but used a processor in order to map the `declaredLicenses` to `SpdxExpression`. 

> Phrased differently, why is homepageUrl not called declaredHomepageUrl. Well, because "declared" is redundant in this (the analyzer's) context.

From this point of view it makes sense to name it "authors" but as you mentioned we have to rename consequently also "declaredLicenses" to "licenses".

But we have to come to a conclusion and therefor I'll try to summarize the suggestions and we should decide on one:

1. Field name `declaredAuthors` and put it before `declaredLicenses`.
2. Field name `declaredAuthors` and put it before `description`.
3. Field name `authors` and put it (?) before `declaredLicenses` or `descriptions`. In addition, rename `declaredLicenses` to `licenses`.

Reply:
I've put this onto the [agenda for the next dev meeting](https://github.com/oss-review-toolkit/ort/wiki/ORT-Developer-Meetings#agenda-for-next-meeting-2021-02-18) to move forward.